statistics_templates.js

```diff
@@ -0,0 +1,136 @@
+/* -----------------------------------------------------------------------------
+ * @copyright (C) 2018, Alert Logic, Inc
+ * @doc
+ *
+ * Statistics functions - to be used from the collectors code,
+ * as base for custom statistics funs
+ *
+ * @end
+ * -----------------------------------------------------------------------------
+ */
+
+const AWS = require('aws-sdk');
+const async = require('async');
+const moment = require('moment');
+
+const m_alAws = require('./al_aws');
+
+const AWS_STATISTICS_PERIOD_MINUTES = 15;
+const MAX_ERROR_MSG_LEN = 1024;
+
+/**
+ * @typedef {Object} Stat
+ * @property {string} Label - status of healtcheck (ok, warning, error).
+ * @property {array} Datapoints - array of datapoints.
+ *
+*/
+
+/**
+ * @typedef {Object} StatError
+ * @property {string} Label - status of healtcheck (ok, warning, error).
+ * @property {string} StatisticsError - description of error happened during getting statistic
+ *
+*/
+
+
+/**
+ * All custom statistic funs should follow the same interface as this function.
+ *
+ * @function
+ *
+ * @param {function} callback - callback, which is called by health check when it's done.
+ *
+ * @returns {function} callback(err, result)
+ *
+ *     {Error}  err             - The Error object if an error occurred, null otherwise
+ *     {Stat|StatError}  result - The Stat object, or StatError if error occured
+ *
+ */
+
+function customStatFunExample(callback) {
```

Comment:
I think it is better to put the sample into comments.

Reply:
@kkuzmin unfortunately, this way it won't appear in the `jsdoc`....